envoy/http/stream_reset_handler.h

```diff
@@ -36,7 +36,9 @@ enum class StreamResetReason {
   // Received payload did not conform to HTTP protocol.
   ProtocolError,
   // If the stream was locally reset by the Overload Manager.
-  OverloadManager
+  OverloadManager,
+  // If stream was locally reset due to HTTP/1 upstream half closing before downstream.
+  Http1PrematureUpstreamHalfClose,
```

Comment:
Yes, the reason is changing. However I can't see where this change would matter in HCM. Can you point to specific place, please?

I can also revert this change and just use local reset flag if you think it would be safer.

Reply:
Not a concern on second look.